docs/draft/transformation-lifecycle/transformation-lifecycle.md

```diff
@@ -0,0 +1,23 @@
+# [WIP] Transformation Lifecycle draft
+
+The basic idea of the TransformationLifecycle is to create a interface that brings all the plugins on a same level.
+The gist of the TransformationLifecycle is to provide the same lifecycle for every transformation, the idea came from the Maven lifecycle. As in the Maven lifecycle we have different phases, in our case so called transformation phases.
+With the lifecycle in mind we can build the TransformationLifecycleInterface containing a method for every build phase.
+The interface is then implemented by a abstract BasePlugin.
+Every specific plugin has to extend the BasePlugin, so the specific plugin is forced to provide a method for every transformation phase.
+
+**Transformation phases:**
+- `validate` - in this phase the plugin provides a list of supported TOSCA types, then the graph that is enqued for transformation will be checked if it contains types the plugin can not handle.
```

Comment:
just a question: 
the plugins support different node-types we defined [here](https://github.com/StuPro-TOSCAna/TOSCAna/blob/master/docs/draft/plugins/supported_types.md). But the node-types are the same for every plugin, the plugin only has to implement the special transformation of this type. Does it make sense to define every node-type in the BasePlugin? So that every plugin has access to the same node-types and not every plugin has to define the nodetypes by themselfes.

Reply:
This is the approach taken by #169 , check it out.